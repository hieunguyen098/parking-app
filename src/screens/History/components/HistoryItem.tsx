import { StyleSheet, Text, View } from 'react-native';
import React from 'react';
import { FontAwesome5 } from '@expo/vector-icons';
import { Entypo } from '@expo/vector-icons';
import { GlobalStyles } from '../../../constants';
import moment from "moment/moment";
const HistoryItem = ({ item, style = null }: any) => {
    const formattedDate = (dateString: string) => {
        return moment(dateString).format('HH:mm DD/MM/YYYY');
    };

    return (
        <View style={[styles.container, style]}>
            <View style={styles.iconContainer}>
                {item.type === 'month-ticket' && (
                    <Entypo name="ticket" size={32} color={GlobalStyles.colors.lightRed} />
                )}

                {item.type === 'parking' && (
                    <FontAwesome5 name="parking" size={40} color={GlobalStyles.colors.primaryOrange} />
                )}
            </View>
            <View style={styles.contentContainer}>
                <Text style={styles.title}>{item.title}</Text>
                <Text style={styles.place}>{item.extraInfo.place}</Text>
                <Text style={styles.time}>{item.time}</Text>
            </View>
            <View style={styles.priceContainer}>
                <Text style={styles.price}>{item.extraInfo.price}</Text>
            </View>
        </View>
    );
};

export default HistoryItem;

const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        backgroundColor: "#FFFFFF",
        flex: 1,
        borderRadius: 8,
        paddingLeft: 16,
        paddingRight: 16,
        marginBottom: 2,
        paddingTop: 16,
        paddingBottom: 16,
        borderBottomColor: GlobalStyles.colors.lightGrey100,
        borderBottomWidth: 1,
        minHeight: 100,
    },
    iconContainer: {
        marginRight: 12,
    },
    contentContainer: {
        flex: 1,
        justifyContent: 'space-between',
    },
    title: {
        fontSize: 20,
        fontWeight: '500',
    },
    place: { color: GlobalStyles.colors.lightGrey },
    time: { color: GlobalStyles.colors.lightGrey, fontSize: 12 },
    priceContainer: {
        justifyContent: 'flex-end',
        marginLeft: 8,
    },
    price: {
        fontSize: 20,
        fontWeight: '500',
    },
});
