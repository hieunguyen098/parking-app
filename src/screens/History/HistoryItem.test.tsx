import React from 'react';
import HistoryItem from './components/HistoryItem';
import { render } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';

// test('renders home', () => {

//     const tree = renderer.create(

//         <HistoryItem item={{
//             id: '15',
//             type: 'parking',
//             title: 'Thanh toán phí gửi xe',
//             place: 'Nhà xe Trường Đại học Bách Khoa Tp.HCM',
//             time: '17:05 - 27/11/2022',
//             price: '-2.000đ',
//         }} />
//     ).toJSON();
//     expect(tree).toMatchSnapshot();
// });

test('test histoy item', () => {
    const component = (
        <NavigationContainer>
            <HistoryItem
                item={{
                    id: '15',
                    type: 'parking',
                    title: 'Thanh toán phí gửi xe',
                    place: 'Nhà xe Trường Đại học Bách Khoa Tp.HCM',
                    time: '17:05 - 27/11/2022',
                    price: '-2.000đ',
                }}
            />
        </NavigationContainer>
    );

    const tree = render(component);
    expect(tree).toMatchSnapshot();
});
