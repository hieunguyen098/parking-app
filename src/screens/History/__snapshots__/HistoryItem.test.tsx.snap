// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test histoy item 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomColor": "#E8E8E8",
        "borderBottomWidth": 1,
        "flex": 1,
        "flexDirection": "row",
        "minHeight": 100,
        "paddingBottom": 16,
        "paddingTop": 16,
      },
      null,
    ]
  }
>
  <View
    style={
      Object {
        "marginRight": 12,
      }
    }
  >
    <Text />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "500",
        }
      }
    >
      Thanh toán phí gửi xe
    </Text>
    <Text
      style={
        Object {
          "color": "#B3B3B3",
        }
      }
    >
      Nhà xe Trường Đại học Bách Khoa Tp.HCM
    </Text>
    <Text
      style={
        Object {
          "color": "#B3B3B3",
          "fontSize": 12,
        }
      }
    >
      17:05 - 27/11/2022
    </Text>
  </View>
  <View
    style={
      Object {
        "justifyContent": "flex-end",
        "marginLeft": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "500",
        }
      }
    >
      -2.000đ
    </Text>
  </View>
</View>
`;
